lib/ansible/modules/source_control/gitlab_webhook.py

```diff
@@ -0,0 +1,432 @@
+            # * state=present, none found, hook_id  : error
+            # * state=absent, none found, hook_id  : error
+            # hook_id set, none found: error (whether state=present or absent)
+            module.exit_json(changed=False, msg='No webhook match', details='Webhook does not exists for this id %d' % hook_id)
+        num_found = 1
+    else:
+        all_remote_webhooks = project.hooks.list()
+        remote_webhook = HookHelper.findByUrl(all_remote_webhooks, url)
+        num_found = len(remote_webhook)
+
+    local_webhook = HookHelper.asApiObject(url, events, hook_id, token)
+    if remote_webhook and state == "absent":
+        if num_found <= 1:
+            # * state=absent, 1 found, solo=yes : remove
+            # * state=absent, 1 found, solo=no  : remove
+            h = HookHelper.delete(project, remote_webhook, local_webhook)
+            module.exit_json(changed=True, result="Successfully deleted webhook %s" % url, webhook=h)
+        else:
+            if not solo:
+                # * state=absent, 2+ found, solo=no  : remove all
+                deleted = HookHelper.delete(project, remote_webhook, local_webhook)
+                module.exit_json(changed=len(deleted) > 0, result="Successfully deleted %d webhooks" % len(deleted))
+            else:
+                # * state=absent, 2+ found, solo=yes : error
+                module.fail_json(msg='Multiple hooks match',
+                                 details="solo=yes but %d webhooks match for url %s" % (num_found, url))
+
+    elif state == "absent":
+        # state=absent,  none found: error
+        module.exit_json(changed=False, result="Webhook does not exists for this url")
+    else:
+        # state == "present"
+        if remote_webhook and num_found > 1:
+            if not solo and not hook_id:
+                # * state=present, 2+ found, solo=no, no hook_id : error
+                # (but matching heuristic could be further improved)
+                module.fail_json(msg='Multiple webhooks match, specify <hook_id>',
+                                 details="%d webhooks match but no hook_id were provided" % num_found)
+            elif solo:
+                # * state=present, 2+ found, solo=yes : error
+                module.fail_json(msg='More than one webhooks match but only one was expected',
+                                 details="%d webhooks match but solo=yes" % num_found)
+            else:
+                module.fail_json(msg='unexpected error')
+
+        if num_found == 1:
+            # Edit, no confusion
+            # * state=present, 1 found, solo=yes : edit
+            # * state=present, 1 found, solo=no  : edit
+            # * state=present, 1 found, hook_id  : edit
+            try:
+                diff = {'before': str(remote_webhook[0].attributes)}
+                h = HookHelper.update(project, local_webhook, remote_webhook[0])
+                diff['after'] = str(h)
+            except:
```

Comment:
Bare `except:` is very rarely (if ever) what should be used. In this case, it means a `ctrl-c` will be caught and Ansible will keep going. I suggest using something more specific (at least `Exception`, which won't catch `ctrl-c`).